### PR TITLE
Non-root building and signing of packages on Debian family

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Allows for building and signing Debian (and Raspbian and latest Ubuntu 18.04) packages on AWS and VirtualBox, which were failing as root user due to changes in latest versions of gpg-agent.  Ubuntu 14.04 and Ubuntu 16.04 are still built as root, due to gpg-agent version supported in those versions.

### What issues does this PR fix or reference?
With recent Debian 9 (9.5 on VirtualBox, from 9.0 on AWS) the gpg-agent does not start correctly if launched by root user, hence the need to build and sign as a non-root user.  This is due to changes in gpg-agent utilizing systemd and the associated Unix standard sockets failing to start correctly if root.  This results in failing to store the passphrase and hence signing failures.  Also with recent changes the key fingerprint is now used in re-signing with debsign.

### Previous Behavior
Signing packages Debian 9.0 and above would fail on AWS, 9.5 would fail on VirtualBox.  Debian 9.2 on OpenNebula and VirtualBox would succeed.

### New Behavior
Signing Debian 9 packages now succeed on AWS when signing as a non-root user.

### Tests written?
No, use of salt-pack, salt-pack-py3 and salt-auto-pack si sufficient testing

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/debbuild.py [hunk 1]
```diff
@@ -264,6 +273,14 @@ def _create_pbuilders(env):
     if env_overrides and not env_overrides.isspace():
         with salt.utils.files.fopen(pbuilderrc, 'a') as fow:
             fow.write(salt.utils.stringutils.to_str(env_overrides))
+    cmd = "chown {0}:{0} {1}".format(runas, pbuilderrc)
```
Comment (cachedout) [inline salt/modules/debbuild.py]: Is there a reason that `file.chown` isn't being used here instead?

File: salt/modules/debbuild.py [hunk 2]
```diff
@@ -314,13 +331,77 @@ def make_src_pkg(dest_dir, spec, sources, env=None, saltenv='base'):
...
+    _create_pbuilders(env, runas)
     tree_base = _mk_tree()
     ret = []
     if not os.path.isdir(dest_dir):
         os.makedirs(dest_dir)
 
+    # ensure directories are writable
+    root_user = 'root'
+    retrc = 0
+    cmd = "chown {0}:{0} {1}".format(runas, tree_base)
```
Comment (cachedout) [inline salt/modules/debbuild.py]: Same question here about `file.chown`

File: salt/modules/debbuild.py (hunk 1 above)
Comment (dmurphy18) [reply salt/modules/debbuild.py]: Went looking for it and missed it in the doc's, will make change

File: salt/modules/debbuild.py (hunk 2 above)
Comment (dmurphy18) [reply salt/modules/debbuild.py]: same as above

**Review by dwoz — APPROVED** [review]

File: salt/modules/debbuild.py (hunk 1 above)
Comment (cachedout) [reply salt/modules/debbuild.py]: Ah, OK. That makes sense.

**Pull request merged**